include/lsst/sphgeom/CompoundRegion.h

```diff
@@ -125,6 +125,7 @@ class UnionRegion : public CompoundRegion {
     using Region::contains;
     bool contains(UnitVector3d const &v) const override;
     Relationship relate(Region const &r) const override;
+    bool isDisjoint(Region const& other) const override;
```

Comment:
Given the precedent, I do think `isDisjointFrom` would be a slightly better name.  But if that leads to lots of complications with overloading and shadowing it may not be worth it.

Or, if the exiting `isDisjointFrom` implementations are all exact rather than conservative, maybe we call this one `isDefinitelyDisjointFrom`?  That's verbose, but it might make it a little more clear that `False` only means "maybe disjoint".

Reply:
Hmm, but `isDisjointFrom` is also "definitely disjoint", would `isDefinitelyDisjointFrom` make people think that `isDisjointFrom` is less definite?

Reply:
Another option - call this new method `overlaps(other)` but return tri-state value from it (yes/no/maybe or `bool | None` on Python side). That way we can avoid confusion with existing methods and also make `maybe` distinguishable from two other answers.

Reply:
I like that idea, though I suspect it might be a lot of work (without much payoff) to make it actually return `true` sometimes.

Reply:
In those cases it's probably OK to return `maybe`, just like current code is probably doing for `DISTICT` in `relate`. C++ does not have a standard class for tri-state, Boost has `tristate` but I guess we do not want to add it as dependency. Maybe simple `enum` should be OK on C++ side.

Reply:
I have replaced it with `overlaps()` which returns an instance of a new `TriState` class. IntersectRegion implementation for that is not very useful as it cannot correctly guess "definitely overlaps" case, but I think we now mostly care about unions.